### PR TITLE
Add incremental ID to the tasks and missions

### PR DESCRIPTION
**Pull request by adamsr123**

### CODE REVIEW

File: controllers/missions.js [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-    await missions.deleteOne({ _id: req.params.id});
+router.delete("/:id", async (req, res) => {
+    await missions.deleteOne({ _id: req.params.id });
     res.sendStatus(200);
 });
 
 router.post('/', async (req, res, next) => {
     try {
         const mission = req.body;
+        debugger;
```
Comment (amitaiblevy) [inline controllers/missions.js:27]: dude

File: controllers/missions.js [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+        mission._id = await getNextId('missions');
```
Comment (amitaiblevy) [inline controllers/missions.js:28]: We said that we wouldn't put it in Id

File: utils/counters.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const idCountersCollection = getTable('id-counter');
+
+const idPrefix = {
+    tasks: "D",
+    missions: "M"
+};
+
+async function getNextId(type) {
+    // Mongo DB has it's own increment method.
+    const { value: { counter: prevCounter } } = await idCountersCollection.findOneAndUpdate({ type },
```
Comment (amitaiblevy) [inline utils/counters.js:12]: Did you make sure that findOneAndUpdate is atomic?

File: controllers/missions.js (hunk 1 above)
Comment (adamsr123) [reply controllers/missions.js:27]: looooolllllll nice of yours

File: utils/counters.js (hunk 3 above)
Comment (adamsr123) [reply utils/counters.js:12]: All write operations in mongo are atomic when we talk of one specific document

File: controllers/missions.js [hunk 4]
```diff
@@ -1,30 +1,32 @@
...
     res.sendStatus(200);
 });
 
 router.post('/', async (req, res, next) => {
     try {
         const mission = req.body;
+        mission.serialNumber = await getNextId('missions');
         await missions.insertOne(mission);
-        res.sendStatus(200);
+        res.send({ id: mission._id });
```
Comment (amitaiblevy) [inline controllers/missions.js:29]: revert this to status200

File: utils/counters.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const { getTable } = require('../dal');
+
+const idCountersCollection = getTable('id-counter');
+
+const idPrefix = {
+    tasks: "D",
```
Comment (amitaiblevy) [inline utils/counters.js:6]: Why D and not T?

**Pull request merged**